src/Http/Headers/src/HeaderUtilities.cs

```diff
@@ -625,34 +625,34 @@ public static StringSegment UnescapeAsQuotedString(StringSegment input)
         {
             input = RemoveQuotes(input);
 
-            // First pass to calculate the size of the InplaceStringBuilder
+            // First pass to calculate the size of the string
             var backSlashCount = CountBackslashesForDecodingQuotedString(input);
 
             if (backSlashCount == 0)
             {
                 return input;
             }
 
-            var stringBuilder = new InplaceStringBuilder(input.Length - backSlashCount);
-
-            for (var i = 0; i < input.Length; i++)
+            return string.Create(input.Length - backSlashCount, input, (span, segment) =>
```

Comment:
Should these delegates be cached to keep it equivalent to the previously allocation free for loop?

Reply:
They don't capture and would get cached but I can make it more explicit too.